### ISSUE TITLE
VS Code filters specific tooling paths from environment Path when it launches a Terminal

### ISSUE THREAD

**Issue by CreepyGnome** [microsoft/vscode#72444]
After installing VS Code on Windows and launching the terminal from within VS Code it filters out specific paths from the environment path.

FIlters: Python, Node, SQL Server, Chocolatey, NPM. 

This is very selective as it is not truncating as they are in various places in the path and some are in the system path and some are in the user path and it is selectively removing them from that path when launching any terminal (CMD, PS, BASH, etc) from VS Code. 

VS Code must be doing something to cause this as it's to selective to be random and it doesn't happen if Windows launches the terminal window or when other tools launch terminal windows.

Versions from VS Code About:
VS Code Version: 1.33.1 (user setup)
Commit: 51b0b28134d51361cf996d2f0a1c698247aeabd8
Date: 2019-04-11T08:27:14.102Z
Electron: 3.1.6
Chrome: 66.0.3359.181
Node.js: 10.2.0
V8: 6.6.346.32
OS: Windows_NT x64 10.0.17763

Steps to Reproduce:

1. Install Python 2.x
2. Install SQL Server 2016 Express (Installed Via VS 2017) 
3. Install Node (node-v11.14.0-x64.msi) 
4. Install Angular (npm install -g @angular/cli)
5. Install VS Code User Setup (VSCodeUserSetup-x64-1.33.1.exe)
6. Open VS Code
5. Open Terminal from VS Code (CTRL+`)
6. Try to run npm in the terminal -- you will get an error that it cannot find it

Does this issue occur when all extensions are disabled?: Yes

**Comment by CreepyGnome** [comment]
Side note this issue is limited ONLY to the VS Code User Setup (VSCodeUserSetup-x64-1.33.1.exe) if you use the system one everything works as expected. However, it defaulted to downloading the user on for some reason as I wasn't offered a choice as I never choose the user specific ones if I can help it.

This is still an issue just more limited now, and not sure why the User one filters out very specific paths but not others.  Plenty of program files paths remained, while other filtered, also it filtered User App data npm and ProgramData as well. Very surgical filtering it seems that shouldn't have anything to do with a user install version of VS Code.

Attached are the paths depending on what launched the terminal.

[TerminalPathFromVsCodeLaunch.txt](https://github.com/Microsoft/vscode/files/3087060/TerminalPathFromVsCodeLaunch.txt)
[TerminalPathFromWindowsLaunch.txt](https://github.com/Microsoft/vscode/files/3087061/TerminalPathFromWindowsLaunch.txt)

**Comment by Tyriar** [comment]
Are you sure you shut down all VS Code windows after installing the new programs? The terminal's environment is based on the VS Code main process which uses whatever environment the first window started with until all VS Code windows close.

**Comment by CreepyGnome** [comment]
Yes, all one of my windows was closed. Also what I explain wouldn't be
caused by that. How would one window have a different path than another if
the path never changed between the opening of the second window?

I had just installed it and only ever had one window open and one instance
running. Discovered it when I tried to do a new project that needed me to
run NPM. I did a lot of trouble shooting to figure out what was happening
and issoliated to a specific type of install the user installer

It's reproducable I uninstalled and reinstalled with the same installer
after redownloading it and it continued and only worked correctly when I
uninstalled and switched to the full system version and got rid of the user
one. This is an issue in the user specific install.
